shared/templates/rsyslog_logfiles_attributes_modify/oval.template

```diff
@@ -90,27 +90,39 @@
   <!-- Define OVAL variable to hold all the various system log files locations
        retrieved from the different rsyslog configuration files
   -->
-  <local_variable id="var_rfg_log_files_paths" datatype="string" version="1" comment="File paths of all rsyslog configuration files">
-    <object_component item_field="subexpression" object_ref="object_rfg_log_files_paths" />
+  <local_variable id="var_rfg_{{{ _RULE_ID }}}_log_files_paths" datatype="string" version="1" comment="File paths of all rsyslog configuration files">
+    <object_component item_field="subexpression" object_ref="object_rfg_{{{ _RULE_ID }}}_log_files_paths" />
   </local_variable>
 
   <!-- Perform the test if all rsyslog system log files are owned by the appropriate group -->
-  <unix:file_test check="all" check_existence="all_exist" id="test_rsyslog_files_groupownership" version="1" comment="System log files are owned by the appropriate group">
-    <unix:object object_ref="object_rsyslog_files_groupownership" />
-    <unix:state state_ref="state_rsyslog_files_groupownership" />
+  <unix:file_test check="all" check_existence="all_exist" id="test_rsyslog_files_{{{ _RULE_ID }}}_ownership" version="1" comment="System log files are owned by the appropriate group">
+    <unix:object object_ref="object_rsyslog_files_{{{ _RULE_ID }}}_ownership" />
+    <unix:state state_ref="state_rsyslog_files_{{{ _RULE_ID }}}_ownership" />
   </unix:file_test>
 
-  <unix:file_object id="object_rsyslog_files_groupownership" comment="Various system log files" version="1">
-    <unix:filepath datatype="string" var_ref="var_rfg_log_files_paths" var_check="at least one" />
+  <unix:file_object id="object_rsyslog_files_{{{ _RULE_ID }}}_ownership" comment="Various system log files" version="1">
+    <unix:filepath datatype="string" var_ref="var_rfg_{{{ _RULE_ID }}}_log_files_paths" var_check="at least one" />
   </unix:file_object>
 
-  <unix:file_state id="state_rsyslog_files_groupownership" version="1">
+  <unix:file_state id="state_rsyslog_files_{{{ _RULE_ID }}}_ownership" version="1">
```

Comment:
This state contains the expected value for `gid` or `uid`.
It seems to be informed in the `parameter` attribute when calling the template.
```
template:
  name: rsyslog_logfiles_attributes_modify
  vars:
    attribute: groupowner
    parameter: file_groupowner_logfiles_value
```
However, I don't see the informed variables used in the OVAL. They are only used in remediation.
Could them be also used here to avoid these conditionals?

I believe we could remove these jinja conditionals about `ATTRIBUTE` and also the jinja conditionals about the `products` in favor of a generic `state` declaration which uses the variable.
Or even simpler and more flexible, we could pass the value directly instead of a variable. Like:
```
template:
  name: rsyslog_logfiles_attributes_modify
  vars:
    attribute: groupowner
    value: 0
    value@ubuntu2204: 4
    ....
```
What do you think?

Reply:
Sure looks better - thanks for the feedback :bow: